hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/googlepubsub/GooglePubSubSenderTarget.java

```diff
@@ -31,38 +42,46 @@ public boolean equals(Object o) {
             return false;
         }
         GooglePubSubSenderTarget that = (GooglePubSubSenderTarget) o;
+
         return Objects.equals(topicName, that.topicName)
-                && Objects.equals(pubSubEndpoint, that.pubSubEndpoint);
+                && Objects.equals(pubSubEndpoint, that.pubSubEndpoint)
+                && Objects.equals(compressionCodec, that.compressionCodec);
```

Comment:
With this change, we will have to create a  `GooglePubSubClient` for each tuple `<topicName, pubSubEndpoint, compressionCodec>` (earlier we need to do so for each `<topicName, pubSubEndpoint>`). AFAIR, we introduced the `GooglePubSubClientsPool` to reduce the footprint of creating `GooglePubSubClient`s, so I suspect that they are fairly expensive in terms of resource consumption.

Potentially, in our case, it will not cause any troubles, but still I wanted to ask two questions:
1. Is it intentional?
2. Have you considered optimizing it somehow?

Reply:
Good point, to keep the pool as small as possible, the client assigned to a subscription endpoint will be used- no matters what compression codec is requested.